packages/cli/src/services/url.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Service } from 'typedi';
+import config from '@/config';
+
+@Service()
+export class UrlService {
+	/** Returns the base URL n8n is reachable from */
+	readonly baseUrl: string;
+
+	constructor() {
+		this.baseUrl = this.generateBaseUrl();
+	}
+
+	/** Returns the base URL of the webhooks */
+	getWebhookBaseUrl() {
+		let urlBaseWebhook = process.env.WEBHOOK_URL ?? this.baseUrl;
+		if (!urlBaseWebhook.endsWith('/')) {
+			urlBaseWebhook += '/';
+		}
+		return urlBaseWebhook;
+	}
+
+	/** Return the n8n instance base URL without trailing slash */
+	getInstanceBaseUrl(): string {
```

Comment:
What's the difference between `n8n base URL` and `n8n instance base URL`? Is "instance" here the editor? Maybe this could be renamed `getEditorBaseUrl`, WDYT?

Reply:
+1

Reply:
I plan to implement all of that [here](https://github.com/n8n-io/n8n/pull/6713).
For now, this PR contains no actual change, and I'd like to merge it as it is to unblock some other branches.